src/index.ts

```diff
@@ -38,6 +38,15 @@ export function viteSvgToWebfont<T extends GeneratedFontTypes = GeneratedFontTyp
     const virtualModuleId = getVirtualModuleId(moduleId);
     const resolvedVirtualModuleId = getResolvedVirtualModuleId(virtualModuleId);
 
+    const inline = options.inline
+        ? <U extends string | undefined>(css: U) =>
+              css?.replace(/url\(".*?\.([^?]+)\?[^"]+"\)/g, (_, type: T) => {
+                  const font = generatedFonts?.[type];
+                  const fontData = typeof font === 'string' ? btoa(font) : font?.toString('base64');
+                  return `url("data:${MIME_TYPES[type]};charset=utf-8;base64,${fontData}")`;
+              }) as U
+        : <U>(css: U) => css;
```

Comment:
3 comments here 🙂 

1. I would prefer we use `processedOptions`, instead of `options` (which means the value needs to be added to `parseOptions`
2. `btoa` is a web-platform API (supported in Node for compatibility), we should prefer `Buffer.from(font, 'base64')`
3. I would rather to avoid the ternary here, to make this easier to read:

```typescript
const inline = <U extends string | undefined>(css: U) => {
    if (!processedOptions.inline) {
        return css;
    }
    return css?.replace(/url\(".*?\.([^?]+)\?[^"]+"\)/g, (_, type: T) => {
        const font = generatedFonts?.[type];
        const fontData = typeof font === 'string' ? btoa(font) : font?.toString('base64');
        return `url("data:${MIME_TYPES[type]};charset=utf-8;base64,${fontData}")`;
     }) as U;
};
```

Reply:
1. I think you have meant `Buffer.from(font).toString('base64')`
2. `Buffer.from` can create from both buffer and string, good call

resolved

Reply:
@atlowChemi, changed in fcb6a1403e66b1c3a83c5d6a558ede9fef787d3b